src/include/Parser.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <Lexer.h>
+#include <ASTree.h>
+#include <exception>
+
+class Parser {
+    Lexer m_lexer;
+    Token m_currentToken;
+
+public:
+    explicit Parser(Lexer lexer);
+    void buildTree();
+private:
+    Token getNextToken() {
+        return m_currentToken = m_lexer.getToken();
```

Comment:
What is this method for? I imagine whoever owns the `Parser` won't need individual tokens from the `Lexer`, and anything inside `Parser` can just call `m_lexer.getToken()` directly.

(Also, I personally find assign-to-instance-variable-*and*-return confusing.)

Reply:
i'm... actually not sure. one of the guides you sent said this was a buffer to allow lookahead, but that doesn't make sense to me anymore lol, I'll double check later

Reply:
In particular I have a feeling @CoffeeTableEspresso has some thoughts on how to do that 😛